cmd/ws/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/go-kit/kit/log"
+	kitprometheus "github.com/go-kit/kit/metrics/prometheus"
+	"github.com/mainflux/mainflux"
+	manager "github.com/mainflux/mainflux/manager/client"
+	adapter "github.com/mainflux/mainflux/ws"
+	"github.com/mainflux/mainflux/ws/api"
+	"github.com/mainflux/mainflux/ws/nats"
+	broker "github.com/nats-io/go-nats"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	defPort       = "8180"
+	defNatsURL    = broker.DefaultURL
+	defManagerURL = "http://localhost:8180"
+	envPort       = "MF_WS_ADAPTER_PORT"
+	envNatsURL    = "MF_NATS_URL"
+	envManagerURL = "MF_MANAGER_URL"
+	topic         = "src.*"
```

Comment:
How do we prevent loopback from messages on the topic `src.ws`, i.e. messages WS client is publishing?

Reply:
It actually makes sense, since 2 websocket clients should be able to “hear” eachother. However, it must be carefully handled to avoid making “self” dos.

Reply:
With of-the-shelf MQTT this is a problem - MQTT loopbacks, but does not keep any "publisher" information, so you can not even drop the message you sent yourself. Here I suppose it will be easier.

Reply:
Currently, I am not sending message to subscriber who is publisher of that message at the same time. I determine if message should be sent by checking publisher id contained within the message.

Reply:
@anovakovic01 yes, I see... OK for now, however we will have to think how we can improve design in the future to avoid these loopbacks, as on every `send` currently we have one additional `receive + ignore`.

Reply:
@drasko I'll try to think of better solution. Of course, I'm open for suggestions.

Reply:
@anovakovic01 @drasko see my second comment, it’s one way built around local state keeping instead of subscribe and filter.

Reply:
I do not have one right now :).

But as I said - situation is even worst with MQTT, as we can change `on_publish` just to limited extent. Plus it is state-full protocol, and these brokers have their way of clustering often.

So, I think we have to find some elegant solution that we will apply to all adapters (all 4 of them face the same issue).

Reply:
@mijicd It's definitely better. By doing that we would avoid receive message from NATS that Drasko mentioned, so only ignore part stays.